app/controllers/item_seasons_controller.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class ItemSeasonsController < ApplicationController
+  before_action :set_item_season, only: %i[show]
+
+  def index
+    @item_seasons = ItemSeason.all
+  end
+
+  def show; end
+
+  private
+
+  def set_item_season
+    @item_season = ItemSeason.find_by_month_index(params[:month_index])
```

Comment:
so, I looked around a bit and from what I understand: metaprogramming is writing programs that write programs. The first one worked because rails generated a bunch of stuff under the hood, but it has a worse perfomance. I read also that it might not be great for readability, searchability and security?